### PR TITLE
fix(core): visible incorrect initial position from `TuiDropdownComponent` & `TuiHintComponent` without animation & ngZoneEventCoalescing

### LINKED ISSUE #3514: 🐞 - ngZoneEventCoalescing without animations causes wrong initial position on some components
### Which @taiga-ui/* package(s) are the source of the bug?

Don't known / other

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/angular-mrvppt-jsouuz

### Is this issue blocking you?

Non-Blocking

### Description

Go to [stackblitz](https://stackblitz.com/edit/angular-mrvppt-jsouuz), open the hint by click or the select.

This is caused by  `ngZoneEventCoalescing: true`.

And disabling the animations.

```ts
{
  provide: TUI_ANIMATIONS_DURATION,
  useFactory: () => 0,
},
```
slowed down video:

icon

input

### Angular version

15.0.0

### Taiga UI version

3.16.0

### Which browsers have you used?

- [X] Chrome
- [X] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [ ] Windows
- [X] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by juanfran**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #3514 

## What is the new behavior?

`TuiHintComponent` & `TuiDropdownComponent` are initially hidden until they have position.

 `TuiDropdownComponent` set an initial width in the component constructor to calculate the initial position.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/core/directives/dropdown/dropdown.component.ts [hunk 1]
```diff
@@ -68,6 +68,8 @@ export class TuiDropdownComponent {
         position$.pipe(takeUntil(destroy$)).subscribe(([top, left]) => {
             this.update(top, left);
         });
+
+        this.updateWidth();
```
Comment (splincode) [inline projects/core/directives/dropdown/dropdown.component.ts:72]: change to `platformBrowserDynamic().bootstrapModule(AppBrowserModule, { ngZoneRunCoalescing: true});` in main.browser.ts please

Comment (splincode) [reply projects/core/directives/dropdown/dropdown.component.ts:72]: wait for it https://github.com/Tinkoff/taiga-ui/pull/3530

**Comment by waterplea** [comment]
I wonder why first screenshot has flipped position 🤔

**Comment by juanfran** [comment]
I have no idea, I'll look into it on Monday

**Comment by waterplea** [comment]
> I have no idea, I'll look into it on Monday

Probably just a flaky test, it's passing now.

**Comment by splincode** [comment]
@waterplea @nsbarsukov @vladimirpotekhin hello, please review

**Review by waterplea — APPROVED** [review]

**Pull request merged**